### PR TITLE
Changed the bodies of Func, For, Branch, Shader to [Expr]

### PR DESCRIPTION
**Pull request by Codysseus**
This ties in with issue #4

### CODE REVIEW

File: src/Parser.hs [hunk 1]
```diff
@@ -357,10 +352,8 @@ parseFunc =
     (params,typ) <- parseFuncSignature
     DT.traceM $ "Func signature parts are " ++ show params ++ " and " ++ show typ
     reservedOp "="
-    reservedOp "{"
-    expr <- parseExpr
-    reservedOp "}"
-    let f = Func name params typ (Body expr "")
+    expr <- braces $ many1 parseExpr
```
Comment (Codysseus) [inline src/Parser.hs:355]: Dang, I missed a call to `parseBlock` here.

File: src/Parser.hs [hunk 2]
```diff
@@ -21,8 +21,11 @@ data PState = PState {
   progs    :: [(String,Prog)]  -- programs parsed along the way
 }
 
-translate :: String -> a -> Parser a
-translate s a = reserved s *> return a
+consume :: (String, a)-> Parser a
+consume (s,a) = reserved s *> return a
```
Comment (montymxb) [inline src/Parser.hs:25]: I *think* this can be `reserved s $> a` to shorten it up.

File: src/Parser.hs [hunk 3]
```diff
@@ -245,26 +243,26 @@ bops = [
...
+  Const <$> (reserved "const" *> parseType) <*> lowIdentifier <*> (reserved "=" *> parseExpr)
   <|>
-  Update <$> (reserved "set" *> lowIdentifier) <*> parseExpr <*> parseExpr
+  Update <$> (reserved "set" *> lowIdentifier) <*> parseExpr
   <|>
-  Out <$> (reserved "out" *> lowIdentifier) <*> parseExpr <*> parseExpr
+  Out <$> (reserved "out" *> lowIdentifier) <*> parseExpr
   <|>
-  Branch <$> (reserved "if" *> parseExpr) <*> (reserved "then" *> (reserved "{" *> parseExpr <* reserved "}")) <*> (reserved "else" *> (reserved "{" *> parseExpr <* reserved "}"))
+  Branch <$> (reserved "if" *> parseExpr) <*> (reserved "then" *> parseBlock) <*> (reserved "else" *> parseBlock)
```
Comment (montymxb) [inline src/Parser.hs:263]: I like breaking this out into blocks, nice!

File: src/Parser.hs [hunk 4]
```diff
@@ -245,26 +243,26 @@ bops = [
...
   <|>
-  For <$> (reserved "for" *> parseExpr) <*> (reserved "do" *> return Nothing) <*> (reserved "{" *> parseExpr <* reserved "}") <*> parseExpr
+  For <$> (reserved "for" *> parseExpr) <*> (consume ("do", Nothing)) <*> parseBlock
```
Comment (montymxb) [inline src/Parser.hs:265]: This is a bit of hack still. It's just a for loop w/ no referencing variable in the loop itself, but the restructuring is fine here. We just need to take care of it before we finish.

File: src/Parser.hs (hunk 1 above)
Comment (montymxb) [reply src/Parser.hs:355]: We can probably add this in here then, would be good to keep the usage consistent.

**Review by montymxb — APPROVED** [review]
This looks pretty good overall. There are a couple of nits, but more stylistic, and we can remedy this down the road. I'll try and put some more work on this later today if i can.

**Pull request merged**